File: app/src/main/java/com/commonpepper/photosen/network/datasource/AbstractListDataSourceFactory.java
package com.commonpepper.photosen.network.datasource;

import androidx.lifecycle.MutableLiveData;
import androidx.paging.DataSource;

public abstract class AbstractListDataSourceFactory<T> extends DataSource.Factory<Integer, T> {

    protected MutableLiveData<AbstractListDataSource<T>> liveDataSource = new MutableLiveData<>();

    public MutableLiveData<AbstractListDataSource<T>> getLiveDataSource() {
        return liveDataSource;
    }
}

File: app/src/main/java/com/commonpepper/photosen/network/datasource/AbstractListDataSourceFactory.kt
package com.commonpepper.photosen.network.datasource

import androidx.lifecycle.MutableLiveData
import androidx.paging.DataSource.Factory

abstract class AbstractListDataSourceFactory<T> : Factory<Int?, T>() {
    var liveDataSource = MutableLiveData<AbstractListDataSource<T>>()
}
